nexuse2e-webapp/src/main/frontend/src/app/data/data.service.ts

```diff
@@ -16,12 +19,47 @@ export class DataService {
     }
   }
 
-  get<T>(path: string): Promise<T> {
+  private get<T>(path: string, cache: boolean): Promise<T> {
+    if (cache) {
+      if (this.cache[path]) {
+        return this.cache[path] as Promise<T>;
+      }
+      this.cache[path] = this.http.get<T>(this.API_URL + path).toPromise();
+    }
     return this.http.get<T>(this.API_URL + path).toPromise<T>();
   }
 
-  // eslint-disable-next-line
-  post(path: string, body: any) {
+  getAllMessages(): Promise<Message[]> {
+    return this.get<Message[]>("/messages", false);
+  }
+
+  getFullUsername(): Promise<string> {
+    return this.get<string>("/full-username", true);
+  }
+
+  getLoggedIn() {
+    // This response should not be cached because otherwise,
+    // the user would not be logged out if their session expired
+    return this.get("/logged-in", false);
+  }
+
+  getMachineName(): Promise<string> {
+    return this.get<string>("/machine-name", true);
+  }
+
+  getVersion(): Promise<string[]> {
+    return this.get<string[]>("/version", true);
+  }
+
+  post(path: string, body: unknown) {
     return this.http.post(this.API_URL + path, body).toPromise();
   }
```

Comment:
Can we make this private and some type info?
```suggestion
  private post(path: string, body: unknown): Promise<void> {
    return this.http.post<void>(this.API_URL + path, body).toPromise();
  }
```